lib/wasm/WebAssemblyParser.js

```diff
@@ -45,34 +45,33 @@ class WebAssemblyParser extends Tapable {
 		const exports = (state.module.buildMeta.providedExports = []);
 		t.traverse(ast, {
 			ModuleExport({ node }) {
-				const moduleExport = /** @type {t.ModuleExport} */ (node);
-				exports.push(moduleExport.name);
+				exports.push(node.name);
 			},
 
 			ModuleImport({ node }) {
-				const moduleImport = /** @type {t.ModuleImport} */ (node);
-
 				let onlyDirectImport = false;
 
-				if (isMemoryImport(moduleImport) === true) {
+				if (isMemoryImport(node) === true) {
 					onlyDirectImport = true;
 				}
 
-				if (isTableImport(moduleImport) === true) {
+				if (isTableImport(node) === true) {
 					onlyDirectImport = true;
 				}
 
 				const dep = new WebAssemblyImportDependency(
-					moduleImport.module,
-					moduleImport.name,
-					moduleImport.descr,
+					node.module,
+					node.name,
+					node.descr,
 					onlyDirectImport
 				);
 
 				state.module.addDependency(dep);
 			}
 		});
 
+		state.module._ast = ast;
```

Comment:
I don't know if this is the best way to add an AST to the module?

Reply:
Create a static WeakMap and store it there. Offer a static `WebAssemblyParser.getDecodedAst()` method to access it from the Generator.